lib/ddtrace/workers/trace_writer.rb

```diff
@@ -199,7 +199,7 @@ def write(trace)
         # reconfigure the worker accordingly.
         # NOTE: It's important we do this before queuing or it will drop the current trace,
         #       because #after_fork resets the buffer.
-        perform
+        perform if async?
```

Comment:
This was creating a background thread for non-async writers.

Reply:
Creating a background thread for non-async writers is still not correct, but the solution was a little bit more involved, given `perform` can change the result of `async?` calls.